### ISSUE TITLE
Yi-34B finetune出错

### ISSUE THREAD

**Issue by chenyzh28** [01-ai/Yi#152]
使用8卡的A800，900G内存，发现模型加载（main.py里的create_hf_model函数）都直接爆内存了，子进程都被kill了。
![image](https://github.com/01-ai/Yi/assets/35193979/0d348916-a865-4079-8c19-92e531afd809)

**Comment by tfal-yan** [comment]
[2023-11-21 20:42:18,931] [INFO] [comm.py:637:init_distributed] cdb=None
[2023-11-21 20:42:22,682] [INFO] [launch.py:315:sigkill_handler] Killing subprocess 156890
[2023-11-21 20:42:22,704] [INFO] [launch.py:315:sigkill_handler] Killing subprocess 156891
[2023-11-21 20:42:22,723] [INFO] [launch.py:315:sigkill_handler] Killing subprocess 156893
[2023-11-21 20:42:22,723] [INFO] [launch.py:315:sigkill_handler] Killing subprocess 156894
[2023-11-21 20:42:22,740] [INFO] [launch.py:315:sigkill_handler] Killing subprocess 156895
我这边用6b也遇到类似问题，请问找到解决方案没？

**Comment by chenyzh28** [comment]
> [2023-11-21 20:42:18,931] [INFO] [comm.py:637:init_distributed] cdb=None [2023-11-21 20:42:22,682] [INFO] [launch.py:315:sigkill_handler] Killing subprocess 156890 [2023-11-21 20:42:22,704] [INFO] [launch.py:315:sigkill_handler] Killing subprocess 156891 [2023-11-21 20:42:22,723] [INFO] [launch.py:315:sigkill_handler] Killing subprocess 156893 [2023-11-21 20:42:22,723] [INFO] [launch.py:315:sigkill_handler] Killing subprocess 156894 [2023-11-21 20:42:22,740] [INFO] [launch.py:315:sigkill_handler] Killing subprocess 156895 我这边用6b也遇到类似问题，请问找到解决方案没？

减少并发数可以解决。

**Comment by tfal-yan** [comment]
请问是sft的脚本里的哪个参数控制并发？多谢

**Comment by chenyzh28** [comment]
> 请问是sft的脚本里的哪个参数控制并发？多谢

deepspeed --include localhost:2,3,4,5,6,7 main.py

**Comment by tfal-yan** [comment]
[2023-11-24 11:25:38,317] [INFO] [real_accelerator.py:158:get_accelerator] Setting ds_accelerator to cuda (auto detect)
[2023-11-24 11:25:39,963] [WARNING] [runner.py:203:fetch_hostfile] Unable to find hostfile, will proceed with training with local resources on
[2023-11-24 11:25:39,963] [INFO] [runner.py:570:main] cmd = /root/anaconda3/envs/testCase/bin/python -u -m deepspeed.launcher.launch --world_isb2NhbGhvc3QiOiBbMSwgMiwgMywgNF19 --master_addr=127.0.0.1 --master_port=29500 --enable_each_rank_log=None main.py --data_path /DATA_PATH/ --moe_or_path /MODEL_PATH/ --per_device_train_batch_size 1 --per_device_eval_batch_size 1 --max_seq_len 4096 --learning_rate 2e-6 --weight_decay 0_train_epochs 4 --training_debug_steps 20 --gradient_accumulation_steps 1 --lr_scheduler_type cosine --num_warmup_steps 0 --seed 1234 --gradiekpointing --zero_stage 2 --deepspeed --offload --lora_dim 128 --lora_module_name layers. --output_dir ./output_Yi_6b_chat_sft_lora
[2023-11-24 11:25:41,933] [INFO] [real_accelerator.py:158:get_accelerator] Setting ds_accelerator to cuda (auto detect)
[2023-11-24 11:25:43,567] [INFO] [launch.py:145:main] WORLD INFO DICT: {'localhost': [1, 2, 3, 4]}
[2023-11-24 11:25:43,567] [INFO] [launch.py:151:main] nnodes=1, num_local_procs=4, node_rank=0
[2023-11-24 11:25:43,567] [INFO] [launch.py:162:main] global_rank_mapping=defaultdict(<class 'list'>, {'localhost': [0, 1, 2, 3]})
[2023-11-24 11:25:43,567] [INFO] [launch.py:163:main] dist_world_size=4
[2023-11-24 11:25:43,567] [INFO] [launch.py:165:main] Setting CUDA_VISIBLE_DEVICES=1,2,3,4
[2023-11-24 11:25:45,728] [INFO] [real_accelerator.py:158:get_accelerator] Setting ds_accelerator to cuda (auto detect)
[2023-11-24 11:25:45,729] [INFO] [real_accelerator.py:158:get_accelerator] Setting ds_accelerator to cuda (auto detect)
[2023-11-24 11:25:45,736] [INFO] [real_accelerator.py:158:get_accelerator] Setting ds_accelerator to cuda (auto detect)
[2023-11-24 11:25:45,740] [INFO] [real_accelerator.py:158:get_accelerator] Setting ds_accelerator to cuda (auto detect)
/root/anaconda3/envs/testCase/lib/python3.11/site-packages/transformers-4.35.0-py3.11.egg/transformers/deepspeed.py:23: FutureWarning: transfoeepspeed module is deprecated and will be removed in a future version. Please import deepspeed modules directly from transformers.integrations
  warnings.warn(
[2023-11-24 11:25:48,513] [INFO] [comm.py:637:init_distributed] cdb=None
[2023-11-24 11:25:48,513] [INFO] [comm.py:668:init_distributed] Initializing TorchBackend in DeepSpeed with backend nccl
/root/anaconda3/envs/testCase/lib/python3.11/site-packages/transformers-4.35.0-py3.11.egg/transformers/deepspeed.py:23: FutureWarning: transfoeepspeed module is deprecated and will be removed in a future version. Please import deepspeed modules directly from transformers.integrations
  warnings.warn(
/root/anaconda3/envs/testCase/lib/python3.11/site-packages/transformers-4.35.0-py3.11.egg/transformers/deepspeed.py:23: FutureWarning: transfoeepspeed module is deprecated and will be removed in a future version. Please import deepspeed modules directly from transformers.integrations
  warnings.warn(
/root/anaconda3/envs/testCase/lib/python3.11/site-packages/transformers-4.35.0-py3.11.egg/transformers/deepspeed.py:23: FutureWarning: transfoeepspeed module is deprecated and will be removed in a future version. Please import deepspeed modules directly from transformers.integrations
  warnings.warn(
[2023-11-24 11:25:48,731] [INFO] [comm.py:637:init_distributed] cdb=None
[2023-11-24 11:25:48,734] [INFO] [comm.py:637:init_distributed] cdb=None
[2023-11-24 11:25:48,746] [INFO] [comm.py:637:init_distributed] cdb=None
[2023-11-24 11:25:50,569] [INFO] [launch.py:315:sigkill_handler] Killing subprocess 1172717
[2023-11-24 11:25:50,594] [INFO] [launch.py:315:sigkill_handler] Killing subprocess 1172718
[2023-11-24 11:25:50,594] [INFO] [launch.py:315:sigkill_handler] Killing subprocess 1172719
[2023-11-24 11:25:50,642] [INFO] [launch.py:315:sigkill_handler] Killing subprocess 1172720
[2023-11-24 11:25:50,661] [ERROR] [launch.py:321:sigkill_handler] ['/root/anaconda3/envs/testCase/bin/python', '-u', 'main.py', '--local_rank=data_path', '/DATA_PATH/', '--model_name_or_path', '/MODEL_PATH/', '--per_device_train_batch_size', '1', '--per_device_eval_batch_size', '1', seq_len', '4096', '--learning_rate', '2e-6', '--weight_decay', '0.', '--num_train_epochs', '4', '--training_debug_steps', '20', '--gradient_acion_steps', '1', '--lr_scheduler_type', 'cosine', '--num_warmup_steps', '0', '--seed', '1234', '--gradient_checkpointing', '--zero_stage', '2'epspeed', '--offload', '--lora_dim', '128', '--lora_module_name', 'layers.', '--output_dir', './output_Yi_6b_chat_sft_lora'] exits with return
我这里还是失败，4张24G的卡【1,2,3,4】，6b，sft，lora
